src/MoBi.Core/Services/SimulationLoader.cs

```diff
@@ -56,8 +56,13 @@ private void addSimulationToProject(IMoBiSimulation simulation, MoBiMacroCommand
          var moBiSimulation = simulation;
 
          var project = _context.CurrentProject;
+         var originalSimulationName = moBiSimulation.Name;
+         if (!_nameCorrector.CorrectName(project.Simulations, moBiSimulation))
+            return;
+
+         if (originalSimulationName != moBiSimulation.Name) //has been renamed
+            renameColidingModules(simulation.Modules, moBiSimulation.Name, project.Modules, originalSimulationName);
```

Comment:
I would move this all down below line 76 as it was before because this changes the behavior slightly now. Instead of renaming only a cloned simulation this will rename the original too. I'm not sure we should change that in this case.